fix sliceserverheaderbuilder's broken ValueExpression selector

GlenDC opened the pull request:
fixes #881

casualjim commented:
Could you add a test, and verify the generated code will do the right thing for items that look like:

```yaml
headers:
  status:
    type: array
    items:
      type: array
      items:
        type: array
        items:
          type: string
          enum:
          - NEW
          - PUBLISHED
          - OPEN
          - FINISHED
```

GlenDC commented:
@casualjim I added the test, and for now it seems to generate invalid code. I suppose you knew that would happen, hence the question.

Will look into a fix for this issue tomorrow and push it together with the updated test.

GlenDC commented:
@casualjim I replaced the original fix with the actual fix.

`ValueExpression` is now added to the `GenHeader`. Also added your test, and checked the generated code also manually. Seems to be be all fine AFAIK.
